### PR TITLE
feat 💥: message operations with InteractionContext

### PR DESCRIPTION
**Pull request by LordOfPolls**
## What type of pull request is this?

- [ ] Non-breaking code change
- [x] Breaking code change
- [ ] Documentation change/addition
- [ ] Tests change

## Description

A nice chunky PR. Primarily this PR does three things: 
- Add support for editing and deleting messages with `InteractionContext`
- Refactor `Message.edit` to use kwargs
- Refactor `Message.delete` to be all round nicer

### This adds support for editing and deleting ephemeral messages. 
Deleting
```python
message = await ctx.send("test", ephemeral=True)
await context.delete(message)
# or
await message.delete(context=ctx)
```
```python
message = await ctx.send("test", ephemeral=True)
await context.edit(message, content="test edit")
# or
await message.edit(content="test edit", context=ctx)
```

### Kwarg edit
`Message.edit` should really have been kwarg only this whole time. I will die on this hill. 

## Changes

- Refactor `message.delete`
- Add HTTP method for interaction message deletes
- Add `InteractionContext.delete`
- Add `InteractionContext.edit`
- Add interaction deletion support to `Message.delete`
- Add interaction edit support to `Message.edit`

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: naff/models/discord/message.py [hunk 1]
```diff
@@ -559,27 +560,29 @@ async def edit(
         if message_data:
             return self._client.cache.place_message_data(message_data)
 
-    async def delete(self, delay: Absent[Optional[int]] = MISSING) -> None:
+    async def delete(self, delay: int = 0, *, context: "InteractionContext") -> None:
```
Comment (silasary) [inline naff/models/discord/message.py]: This isn't as optional as you'd like it to be, and tests are failing because of it

Comment (LordOfPolls) [reply naff/models/discord/message.py]: whoops. excuse me while i shame myself for that one

Comment (LordOfPolls) [reply naff/models/discord/message.py]: Resolved in bc3e79220e57a97173368691b4b764a170a8a2e0

**Comment by LordOfPolls** [comment]
Test fail related to check changes upstream. This pr passes all checks

**Review by silasary — APPROVED** [review]

**Pull request merged**